idaes/core/util/model_diagnostics.py

```diff
@@ -1041,10 +1219,680 @@ def report_numerical_issues(self, stream=stdout):
+                else:
+                    # This variable does not appear in any constraint
+                    return Constraint.Skip
+
+        else:
+            J = self.jacobian
+
+            def eq_degenerate(m_dh, v):
+                if np.sum(np.abs(J[:, v])) > self.config.tolerance:
+                    return sum(J[c, v] * m_dh.nu[c] for c in m_dh.C) == 0
+                else:
+                    # This variable does not appear in any constraint
+                    return Constraint.Skip
+
+        m_dh.degenerate = Constraint(m_dh.V, rule=eq_degenerate)
+
+        # When y_pos = 1, nu >= m_small
+        # When y_pos = 0, nu >= - m_small
+        def eq_pos_lower(b, c):
+            return b.nu[c] >= -m_small + 2 * m_small * b.y_pos[c]
+
+        m_dh.pos_lower = Constraint(m_dh.C, rule=eq_pos_lower)
+
+        # When y_pos = 1, nu <= M + m_small
+        # When y_pos = 0, nu <= m_small
+        def eq_pos_upper(b, c):
+            return b.nu[c] <= M * b.y_pos[c] + m_small
+
+        m_dh.pos_upper = Constraint(m_dh.C, rule=eq_pos_upper)
+
+        # When y_neg = 1, nu <= -m_small
+        # When y_neg = 0, nu <= m_small
+        def eq_neg_upper(b, c):
+            return b.nu[c] <= m_small - 2 * m_small * b.y_neg[c]
+
+        m_dh.neg_upper = Constraint(m_dh.C, rule=eq_neg_upper)
+
+        # When y_neg = 1, nu >= -M - m_small
+        # When y_neg = 0, nu >= - m_small
+        def eq_neg_lower(b, c):
+            return b.nu[c] >= -M * b.y_neg[c] - m_small
+
+        m_dh.neg_lower = Constraint(m_dh.C, rule=eq_neg_lower)
+
+        # Absolute value
+        def eq_abs_lower(b, c):
+            return -b.abs_nu[c] <= b.nu[c]
+
+        m_dh.abs_lower = Constraint(m_dh.C, rule=eq_abs_lower)
+
+        def eq_abs_upper(b, c):
+            return b.nu[c] <= b.abs_nu[c]
+
+        m_dh.abs_upper = Constraint(m_dh.C, rule=eq_abs_upper)
+
+        # At least one constraint must be in the degenerate set
+        m_dh.degenerate_set_nonempty = Constraint(
+            expr=sum(m_dh.y_pos[c] + m_dh.y_neg[c] for c in m_dh.C) >= 1
+        )
+
+        # Minimize the L1-norm of nu
+        m_dh.obj = Objective(expr=sum(m_dh.abs_nu[c] for c in m_dh.C))
+
+        self.candidates_milp = m_dh
+
+    def _solve_candidates_milp(self, tee: bool = False):
+        """Solve MILP to generate set of candidate equations
+
+        Arguments:
+
+            tee: print solver output (default = False)
+
+        """
+        _log.info("Solving Candidates MILP model.")
+
+        eq_con_list = self.nlp.get_pyomo_equality_constraints()
+
+        results = self._get_solver().solve(self.candidates_milp, tee=tee)
+
+        self.degenerate_set = {}
+
+        if check_optimal_termination(results):
+            # We found a degenerate set
+            for i in self.candidates_milp.C:
+                # Check if constraint is included
+                if self.candidates_milp.abs_nu[i]() > self.config.m_small * 0.99:
```

Comment:
As a general rule, it would be better to make the embedded constant (0.99) a named attribute (at the module scope?)